source/plugin/placeholders/retrievingtext.rst

```diff
@@ -0,0 +1,84 @@
+===================================
+Retrieving and Parsing Placeholders
+===================================
+
+.. javadoc-import::
+    org.spongepowered.api.service.placeholder.PlaceholderService
+    org.spongepowered.api.service.placeholder.PlaceholderParser
+    org.spongepowered.api.service.placeholder.PlaceholderText
+    org.spongepowered.api.service.placeholder.PlaceholderText.Builder
+    org.spongepowered.api.text.Text
+    org.spongepowered.api.text.TextRepresentable
+    org.spongepowered.api.text.channel.MessageReceiver
+
+
+Obtaining A PlaceholderParser
+=============================
+
+There are two ways to obtain a registered :javadoc:`PlaceholderParser`:
+
+* Directly obtain the parser from the Sponge Registry by calling 
+  ``Sponge.getRegistry().getType(PlaceholderParser.class, id);``; or
+* Obtain the parser via :javadoc:`PlaceholderService#getParser(String)`.
+
+In general, we recommend that the second option is used, as plugin-provided implementations may allow for shorthand 
+tokens to resolve to a parser, such as accepting ``name`` as a token, retrieving the parser for ``sponge:name``, 
+thereby allowing for a consistent experience across entire servers.
+
+Creating Text From A PlaceholderParser
+======================================
+
+A :javadoc:`PlaceholderParser` requires a :javadoc:`PlaceholderText` in order to generate an appropriate :javadoc:`Text`
+object. :javadoc:`PlaceholderText {PlaceholderTexts}` can be created by using a :javadoc:`PlaceholderText.Builder` 
+obtained from :javadoc:`PlaceholderService#placeholderBuilder()`.
+
+The builder, as a minimum, allows for the following context to be provided:
+
+* The :javadoc:`PlaceholderParser` to use to generate the :javadoc:`Text` (this must be provided)
+* An associated :javadoc:`MessageReceiver`, allowing for the placeholder to modify its output
+* An argument string that the :javadoc:`PlaceholderParser` can parse, often provided by templating engines
+
+If the service is not the Sponge provided service, the builder may be able to provide more context, though this will 
+require depending on the service providing plugin in question.
```

Comment:
"on the plugin that provides the service".

Reply:
Service has done. It's making it really easy to deal with these comments!